### PR TITLE
[stdlib] Cleanup sort module

### PR DESCRIPTION
**Pull request by mzaks**
Just a few renaming and removal of the Comparable elements list sorting section as it can be delegated to other sort function

### CODE REVIEW

File: stdlib/src/builtin/sort.mojo [hunk 1]
```diff
@@ -210,15 +215,16 @@ fn _quicksort[
 # ===----------------------------------------------------------------------===#
 # partition
 # ===----------------------------------------------------------------------===#
+@always_inline
```
Comment (laszlokindrat) [inline stdlib/src/builtin/sort.mojo:218]: +1 on always inlining the hidden methods, but I'm concerned about potential code bloat with decorating the public APIs. Did you benchmark the effects of these, and if so, did you find a perf reason for doing this?

Comment (mzaks) [reply stdlib/src/builtin/sort.mojo]: Totally agree with you. The implication of code bloat slipped my mind. I removed the always inline decorators from public API.

**Review by laszlokindrat — APPROVED** [review]
Makes sense, thanks for fixing this!

**Comment by laszlokindrat** [comment]
!sync

**Comment by soraros** [comment]
It's probably too late. I think we should make the element type always inferred:
```mojo
sort[T: Comparable, cmp: fn(T, T) -> Bool](x: S[T]): ...
```
to
```mojo
sort[T: Comparable, //, cmp: fn(T, T) -> Bool](x: S[T]): ...
```

**Comment by soraros** [comment]
@laszlokindrat Related, is there any reason why a generic `_cmp_fn_type` is used instead of `fn(type, type) -> Bool`?

**Pull request closed without merging**